src/cmd-buildupload

```diff
@@ -164,8 +160,21 @@ def s3_check_exists(bucket, key):
 
 @retry(stop=retry_stop, retry=retry_boto_exception, retry_error_callback=retry_callback)
 def s3_copy(src, bucket, key, max_age, acl, extra_args={}, dry_run=False):
-    if key.endswith('.json') and 'ContentType' not in extra_args:
-        extra_args['ContentType'] = 'application/json'
+    extra_args = dict(extra_args)
+    if 'ContentType' not in extra_args:
+        if key.endswith('.json'):
+            extra_args['ContentType'] = 'application/json'
+        elif key.endswith('.tar'):
+            extra_args['ContentType'] = 'application/x-tar'
+        elif key.endswith('.xz'):
+            extra_args['ContentType'] = 'application/x-xz'
+        elif key.endswith('.gz'):
+            extra_args['ContentType'] = 'application/x-gzip'
+    if 'ContentEncoding' not in extra_args:
+        if key.endswith('.xz'):
+            extra_args['ContentEncoding'] = 'xz'
```

Comment:
This isn't standardized, so I'd guess it should be `x-xz` if we do it at all.  I'd be inclined to just drop it.

Reply:
Yeah, I was looking at https://en.wikipedia.org/wiki/HTTP_compression#Content-Encoding_tokens. Though at least `curl` doesn't support it, so yeah let's just drop it.